src/main/scala/wdl4s/values/WdlString.scala

```diff
@@ -12,21 +12,29 @@ case class WdlString(value: String) extends WdlPrimitive {
     case r: WdlInteger => Success(WdlString(value + r.value))
     case r: WdlFloat => Success(WdlString(value + r.value))
     case r: WdlFile => Success(WdlString(value + r.value))
+    case WdlOptionalValue(_, Some(r)) => add(r)
+    case r: WdlOptionalValue => emptyValue(r)
     case _ => invalid(s"$value + $rhs")
   }
 
   override def equals(rhs: WdlValue): Try[WdlBoolean] = rhs match {
     case r: WdlString => Success(WdlBoolean(value == r.value))
+    case WdlOptionalValue(_, Some(r)) => equals(r)
+    case r: WdlOptionalValue => emptyValue(r)
     case _ => invalid(s"$value == $rhs")
   }
 
   override def lessThan(rhs: WdlValue): Try[WdlBoolean] = rhs match {
     case r: WdlString => Success(WdlBoolean(value < r.value))
+    case WdlOptionalValue(_, Some(r)) => lessThan(r)
+    case r: WdlOptionalValue => emptyValue(r)
```

Comment:
I'm sure there's some clever way to DRY all these cases, e.g.
```
def operatorIfSet[A, B](wdlOptional: WdlOptionalValue, ifSet: A=> B) = wdlOptional match {
  case WdlOptionalValue(_. Some(a)) => ifSet(a)
  else emptyValue(wdlOptional)
}
```
and then in this situation:
```
override def lessThan[...] {
  [...]
  case o: WdlOptionalValue => operatorIfSet(o, lessThan _)
```